front/src/common/authorization/roleBaseAccessControl.ts

```diff
@@ -0,0 +1,58 @@
+import type { BuiltinRole } from 'common/api/osrdEditoastApi';
+
+export type RequiredUserRolesFor = {
+  VIEWS: {
+    OPERATIONAL_STUDIES: BuiltinRole[];
+    STDCM: BuiltinRole[];
+    INFRA_EDITOR: BuiltinRole[];
+    MAP: BuiltinRole[];
+    ROLLING_STOCK_EDITOR: BuiltinRole[];
+  };
+  USER_PROFILE: {
+    STDCM: BuiltinRole[];
+    OPERATIONAL_STUDIES: BuiltinRole[];
+  };
+  FEATURES: {
+    ACCESS_STDCM_DEBUG: BuiltinRole[];
+    CREATE_NEW_PROJECT_STUDY_SCENARIO: BuiltinRole[];
+  };
+};
+
+export const REQUIRED_USER_ROLES_FOR: RequiredUserRolesFor = {
```

Comment:
This feels overkill since we only want to check if the user is:
- Superuser
- STDCM
- Operational study
- Guest

> [!NOTE]
> The role system is temporary and will be drop soon.